src/Native/Unix/System.Native/pal_mount.c

```diff
@@ -33,7 +33,13 @@ static int32_t GetMountInfo(MountPointFound onFound)
 #else
     struct statvfs* mounts = NULL;
 #endif
+#if defined(__FreeBSD__)
+    // second argument of getmntinfo is mode, which, in FreeBSD, can either be MNT_WAIT (1) or MNT_NOWAIT (2)
+    // check getmntinfo(3) and getfsstat(2) for more info
+    int count = getmntinfo(&mounts, MNT_WAIT);
+#elif
     int count = getmntinfo(&mounts, 0);
+#endif
```

Comment:
The BSD docs state:
```
If no flags are provided, MNT_WAIT is assumed.
```
Presumably then when this code was written it was relying on that, hence the `0`.  But given that it would seem the intention was to use `MNT_WAIT`, do we actually need to special-case this for FreeBSD with an ifdef? or could this PR just be the one-line change to pass `MNT_WAIT` instead of `0`?  Is `MNT_WAIT` not defined on macOS?

Reply:
`MNT_WAIT` is defined on macOS, just checked it and changed code accordingly.

Reply:
However, the `MNT_WAIT` assumption seems to be no longer the case by [this](https://www.freebsd.org/cgi/man.cgi?query=getfsstat&sektion=2&apropos=0&manpath=FreeBSD+11.2-RELEASE+and+Ports).